src/lib/controllers/texts-controller.js

```diff
@@ -0,0 +1,82 @@
+// import { v4 as uuidv4 } from 'uuid'
+
+import Alignment from '@/lib/data/alignment'
+import DownloadController from '@/lib/controllers/download-controller.js'
+import UploadController from '@/lib/controllers/upload-controller.js'
+
+export default class TextsController {
+  createAlignment (originDocSource) {
+    this.alignment = new Alignment(originDocSource)
+  }
+
+  updateOriginDocSource (originDocSource) {
+    if (!this.alignment) {
+      this.createAlignment(originDocSource)
+    } else {
+      this.alignment.updateOriginDocSource(originDocSource)
+    }
+  }
+
+  updateTargetDocSource (targetDocSource) {
+    if (!this.alignment) {
+      this.createAlignment()
+    } else {
+      this.alignment.updateTargetDocSource(targetDocSource)
+    }
+  }
+
+  get originDocSource () {
+    return this.alignment.originDocSource
+  }
+
+  get targetDocSource () {
+    return this.alignment.targetDocSource
+  }
+
+  uploadDocSourceFromFile (fileData) {
+    const uploadType = 'plainSourceUploadFromFile'
+
+    const result = UploadController.upload(uploadType, fileData)
+
+    this.updateOriginDocSource(result.originDocSource)
+    this.updateTargetDocSource(result.targetDocSource)
+  }
+
+  downloadData () {
+    const downloadType = 'plainSourceDownload'
+    const data = {
+      originDocSource: this.originDocSource,
+      targetDocSource: this.targetDocSource
+    }
+    return DownloadController.download(downloadType, data)

```

Comment:
We could have here the following errors:
  - we could define wrong downoad type, it would print the error to console
  - we could start do download before any text would be uploaded - I missed this case - will fix

Reply:
fixed